docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,190 @@
+that perpetually keeping data could result in capacity issues due to the rapid
+increase in data being stored. An example being we now generate ~430,000
+`DigestRunSubscriber` records each day which is about eight times the amount
+compared to that of ten months ago.
+
+We've also gained more insight into what data we're not using for analytics.
+The original plan documented in the previous retention policy was to retain as
+much data as possible to enable us to do as much analysis as needed in the
+future. However, much of this data is never used for analysis and so doesn't
+serve a purpose in the email system.
+
+Contrary to what was previously thought, storing lots of unused data
+indefinitely actually complicates and pollutes our auditing and analysis. An
+example of this can be seen if we look at subscriber lists. We have over
+~73,000 lists with active subscribers but in the last week alone, we've
+accumulated 1139 lists which don't have any subscriptions at all (ended or
+active). These unused lists prevent us from doing doing simple auditing tasks
+easily, like determining which lists are actually subscribed to.
+
+Knowing the above, we thought about how we could cleanup this unused data to
+keep EmailAlertApi's database manageable going forward. There are a couple of
+different approaches to consider, including storing unused data in cloud
+storage away from the systems database (like we do when archiving emails) and
+subsequently deleting the data locally. Or, setting up a means of removing the
+data we don't use from the system entirely.
+
+[archived]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[deleted]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_deletion_worker.rb
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md
+[migrate subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+## Decision
+
+We decided that historic, unused data should be removed periodically from the
+system entirely. It was concluded that the data the system no longer uses
+should only exist for a maximum of one year before removal. We sometimes
+analyse historic data from up to a year ago but we have no evidence of needing
+this data for analytics beyond one year. There is however some bespoke
+behaviour for subscriber lists where we delete certain lists sooner than a year
+but that will be elaborated on in a later section.
+
+We chose to remove the historic data entirely instead of retaining it through
+archiving as the latter option seems unnecessary. If we've deemed the data as
+historic and don't see a use for it (analytical or otherwise) going forward,
+then retaining it via an external data store seems a waste of resources and
+energy in maintaining the process.
+
+It was also concluded that we don't need to change the retention for emails
```

Comment:
```suggestion
Emails (and their associated SubscriptionContents) are the one place where we already manage retention - these are held locally for 7 days, and also archived to Athena indefinitely. We still need this low retention for Emails in order to manage the size of the table. However, the archiving to Athena is inconsistent with the above decision to delete unused data entirely. While we originally thought Athena might serve as a data source for analytics, this has not happened in practice, so the archive data stored there is entirely unused. We therefore decided to remove the archiving mechanism for Emails, since it had no value.
```

Reply:
Saying this is sufficient contradicts the above decision and the context, where we've said that we won't hold unused data indefinitely. Although we could just be more specific and say "sufficient to keep the local database manageable", this is also the best time to clarify the future of archiving in Athena.

Sorry I didn't spot this earlier 😬. We'll need an extra consequence to reflect this. I'd suggest something like the following:

---

### Deprecate archiving Email to Athena

We planned work to remove it in the future. In the meantime, we consider archiving to Athena to be deprecated, and have updated [the documentation around it](https://github.com/alphagov/email-alert-api/blob/1a25d44126709bf8a98955d7609ced439095dd9c/docs/analytics.md) to reflect this status.

Reply:
> Saying this is sufficient contradicts the above decision and the context, where we've said that we won't hold unused data indefinitely.

This was referring to the email deletion mechanism, rather than the archiving aspect. Although they're interlinked so I can see this could be seen as contradictory.

I've updated to your recommendation and added the consequence.